common/semaphore/semaphore.go

```diff
@@ -35,6 +39,16 @@ func (s Semaphore) Acquire(ctx context.Context) error {
 	}
 }
 
+// AcquireNow acquires a permit and returns an error if the buffer is full.
+func (s Semaphore) AcquireNow() error {
```

Comment:
Nit. 
```suggestion
func (s Semaphore) TryAcquire() bool {
```
As in sync package, we can name it `TryAcquire()` and return `bool` instead of an `error`. I don't have a strong opinion on this and hence, I leave it to you.

Reply:
Changed to TryAcquire.